force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -102,16 +110,18 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer countVowels(String str) {
         // Initialize the result Integer
-
-        /*
-        for () {
+        Integer result = 0;
+        
+        for (Integer i = 0, length = str.length(); i < length; i++) {
             // get individual characters from the string - hint use substring or split
+            String subString = str.substring(i, i + 1);
             // check if the character is a vowel
+            if (subString == 'a' || subString == 'e' || subString == 'i' || subString == 'o' || subString == 'u') {
+                result++;
+            }
             // if it is a vowel, increment the result Integer
         }
-        */
-
-        return null; // Replace null with the variable you used to store the result
+        return result; // Replace null with the variable you used to store the result
     }
```

Comment:
If you are checking  for  membership the easiest way to use use a set.  Creating a new string on each iteration is also possibly an expensive operation.  But I never bench marked it. I split mine into a list from the string. 

```
 public static Integer countVowels(String str) {
        // Initialize the result Integer
        Integer vowelCount = 0;
        Set<String> vowels = new Set<String>{
                'a', 'e', 'i', 'o', 'u'
        };
        List<String> characters = str.split('');
        for (String character : characters) {
            if (vowels.contains(character.toLowerCase())) {
                vowelCount++;
            }
        }
        return vowelCount;
    }
 ```